File: app/src/main/java/com/kota/Bahamut/Command/TelnetCommand.java
package com.kota.Bahamut.Command;

import com.kota.Bahamut.ListPage.TelnetListPage;
import com.kota.Bahamut.ListPage.TelnetListPageBlock;

public abstract class TelnetCommand implements BahamutCommandDefs {
    public int Action = 0;
    private boolean _is_done = false;
    public boolean recordTime = true;

    public abstract void execute(TelnetListPage telnetListPage);

    public abstract void executeFinished(TelnetListPage telnetListPage, TelnetListPageBlock telnetListPageBlock);

    public boolean isDone() {
        return this._is_done;
    }

    public void setDone(boolean done) {
        this._is_done = done;
    }

    public boolean isOperationCommand() {
        return true;
    }
}

File: app/src/main/java/com/kota/Bahamut/Command/TelnetCommand.kt
package com.kota.Bahamut.Command

import com.kota.Bahamut.ListPage.TelnetListPage
import com.kota.Bahamut.ListPage.TelnetListPageBlock

abstract class TelnetCommand : BahamutCommandDefs {
    @JvmField
    var Action: Int = BahamutCommandDefs.Companion.LoadBlock
    var isDone: Boolean = false
    @JvmField
    var recordTime: Boolean = true

    abstract fun execute(telnetListPage: TelnetListPage?)

    abstract fun executeFinished(
        telnetListPage: TelnetListPage?,
        telnetListPageBlock: TelnetListPageBlock?
    )

    open val isOperationCommand: Boolean
        get() = true
}
